### ISSUE TITLE
`bower install` seems to ignore shrinkwrap file

### ISSUE THREAD

**Issue by soenkehahn** [shyiko/bower-shrinkwrap-resolver#6]
I'm not sure I fully understand how `bower-shrinkwrap-resolver` is supposed to work, but I can't get it to actually freeze dependencies.

Our project has what should be an indirect dependency listed as a direct dependency in the `bower.json` file. This is to pin down that dependency by version. I added `bower-shrinkwrap-resolver` as described in the README and it successfully created a `bower-shrinkwrap.json` file that seems to pin down all dependencies by exact version. But now when I remove the above-mentioned dependency and do `bower install` it'll just go ahead and install a newer version of that dependency. I would expect `bower-shrinkwrap-resolver` to stick to the old version.

I've tested this with the bower versions `1.7.7`, `1.7.9` and `1.8.0`.

**Comment by shyiko** [comment]
@soenkehahn Any chance you can provide `bower.json` you used for testing?

**Comment by soenkehahn** [comment]
Just tested this with this `bower.json`:

``` json
{
  "name": "my-name",
  "dependencies": {
    "auth0.js": "7.6.1",
    "angular-auth0": "1.0.3"
  }
}
```

If you do `bower install` it'll have `auth0.js#7.6.1`. After removing the `auth0.js` line `bower install` will install `auth0.js#8.0.4`.

**Comment by shyiko** [comment]
Thank you. 

To me it looks like an incorrect behavior on `bower` part:
1. `bower` calls [releases("https://github.com/auth0/auth0.js.git")](https://github.com/shyiko/bower-shrinkwrap-resolver/blob/master/src/index.js#L88) -> `bower-shrinkwrap-resolver` responds with `["7.6.1"]`
2. `bower` calls [fetch({"name":"auth0.js","target":"7.6.1","source":"https://github.com/auth0/auth0.js.git","registry":true}, {"releases":[{"target":"7.6.1","version":"7.6.1"}],"endpoint":{"name":"auth0.js","source":"https://github.com/auth0/auth0.js.git","target":"8.0.4"},"release":"8.0.4","version":"8.0.4","resolution":{}})](https://github.com/shyiko/bower-shrinkwrap-resolver/blob/master/src/index.js#L126) where cached value populated with version not present among "resolved".

One implication is that `bower i --force` should always yield correct result (because cache is not used). Another - not doing [this](https://github.com/shyiko/bower-shrinkwrap-resolver/blob/master/src/index.js#L139) when endpoint.target and cached.endpoint.target are different should also make things right (which is really just a workaround because bower won't hit cache in this case even though it should).

**Comment by shyiko** [comment]
Fixed in 0.4.2.
